source/event/src/TG4TrackManager.cxx

```diff
@@ -55,6 +62,16 @@ TG4TrackManager::TG4TrackManager()
       "Cannot create two instances of singleton.");
   }
 
+#ifdef USE_G4ROOT
```

Comment:
Add initialization to 0 for added data members (see below) to the initialization list;
check if TG4RootNavMgr::GetInstance() returns 0 and does not create an
intsance when selected geometry mode is other than rootGeom
ifdef USE_G4ROOT
  TG4RootNavMgr* fRootNavMgr; ///< Pointer to RootNavMgr to communicate
                              ///< geometry states aka TGeoBranchArrays
  TMCManager* fMCManager;     ///< Cached primarily to restore geometry
#endif

Reply:
Could you cross-check whether the change I made is satisfying?